tuning/.pre-commit-config.yaml

```diff
@@ -0,0 +1,39 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.6.0
+    hooks:
+      - id: trailing-whitespace
+        files: ^tuning/
+      - id: end-of-file-fixer
+        files: ^tuning/
+      - id: check-yaml
+        files: ^tuning/
+      - id: check-added-large-files
+        files: ^tuning/
+
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.11.1
+    hooks:
+      - id: mypy
+        name: mypy type check Python files in 'tuning'
+        language_version: python3.10.12
+        files: ^tuning/
+        args: ['--config-file', 'tuning/.mypy.ini']
+        types: [python]
+
+  - repo: local
+    hooks:
+      - id: run-tuner-pytest
+        name: run pytest on specific tuning files
+        entry: python tuning/run_tuner_pytest.py
+        language: system
+        types: [python]
+        files: ^tuning/
```

Comment:
Have you found other projects running pytest as a pre-commit check? If so, I'm curious to see their setups.

I typically separate _linting_ from _testing_.
* Linting checks _code style_ and formatting and must be _fast_ - 10 seconds at most even for a large project. Lint checks must be completely reliable. Lint checks also have minimal dependencies and are usually static analyses - no code needs to be built or installed besides the check tools themselves.
* Testing checks _code correctness_ and can be significantly slower. Fast tests are obviously better than slow tests, but a mature project can have 10s/100s/1000s of tests. Tests run the project code and can depend on other libraries.

One of the goals of pre-commit is being able to run on _every commit action_ (locally and on CI), and I'd really hesitate to put unit tests next to lint checks. Limiting the check to only run on python files under the tuning/ directory does help, but this would still run unit tests locally when just committing a comment change.

That all being said, if you want to keep tests in the pre-commit config, this probably doesn't do what you want. Pre-commit does its own `git diff` to determine which files to run on and then (if the `pass_filenames` option is `true`, which it is by default: https://pre-commit.com/#hooks-pass_filenames) it will call your script with the file name as an argument. So I think it would run `python tuning/run_tuner_pytest.py tuning/foo.py`, then if you edit multiple .py files it would call the script multiple times, once per each file.